dbms/tests/queries/0_stateless/00726_read_in_primary_key_order.sql

```diff
@@ -0,0 +1,12 @@
+CREATE DATABASE IF NOT EXISTS test;
+
+DROP TABLE IF EXISTS test.merge1;
+
+CREATE TABLE test.merge1(a Date, b UInt64) ENGINE=MergeTree() PARTITION BY a ORDER BY b;
+
+INSERT INTO test.merge1 VALUES (1, 1), (1, 2), (1, 3), (100, 1), (100, 2), (200, 1), (200, 4);
```

Comment:
It doesn't test for a table with multiple parts.

Reply:
It doesn't test for a stream with multiple blocks.

Reply:
It doesn' test for case when one part is read by multiple threads.

Reply:
> It doesn' test for case when one part is read by multiple threads.

Hmm, how would this test look like?